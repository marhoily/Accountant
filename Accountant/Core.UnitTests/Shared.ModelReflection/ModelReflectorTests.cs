using ApprovalTests;
using ApprovalTests.Reporters;
using NewModel.Accounting.Persistence;
using NewModel.Shared.ModelReflection;
using Newtonsoft.Json;
using NUnit.Framework;

namespace NewModel.UnitTests.Shared.ModelReflection
{
    [TestFixture]
    [UseReporter(typeof(AraxisMergeReporter))]
    public sealed class ModelReflectorTests
    {
        ModelReflector mUnderTest;

        [SetUp]
        public void SetUp()
        {
            mUnderTest = new ModelReflector();
        }

        [Test]
        public void VerifyReflectLedger()
        {
            var entries = mUnderTest.Reflect(typeof(Ledger));
            Approvals.Verify(
                JsonConvert.SerializeObject(entries, Formatting.Indented));
        }
        [Test]
        public void VerifyReflectBatch()
        {
            var entries = mUnderTest.Reflect(typeof(Batch));
            Approvals.Verify(
                JsonConvert.SerializeObject(entries, Formatting.Indented));
        }
    }

}