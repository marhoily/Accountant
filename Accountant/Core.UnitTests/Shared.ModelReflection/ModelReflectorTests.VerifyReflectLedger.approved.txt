[
  {
    "HasKey": true,
    "Name": "Account",
    "Properties": [
      {
        "IsNavigation": false,
        "Definition": "public string Name { get; set; }",
        "Initializer": "Name = value.Name",
        "Finalizer": "Name = Name"
      },
      {
        "IsNavigation": false,
        "Definition": "public string Description { get; set; }",
        "Initializer": "Description = value.Description",
        "Finalizer": "Description = Description"
      },
      {
        "IsNavigation": false,
        "Definition": "public AccountOptions Options { get; set; }",
        "Initializer": "Options = value.Options",
        "Finalizer": "Options = Options"
      },
      {
        "IsNavigation": false,
        "Definition": "public AccountModifiers Modifier { get; set; }",
        "Initializer": "Modifier = value.Modifier",
        "Finalizer": "Modifier = Modifier"
      },
      {
        "IsNavigation": false,
        "Definition": "public DateTime LastRevisionDate { get; set; }",
        "Initializer": "LastRevisionDate = value.LastRevisionDate",
        "Finalizer": "LastRevisionDate = LastRevisionDate"
      },
      {
        "IsNavigation": true,
        "Definition": "public string Parent { get; set; }",
        "Initializer": "Parent = value.Parent == null ? null : value.Parent.Name",
        "Finalizer": "Parent = Parent == null ? null : context.AccountByKey[Parent]"
      },
      {
        "IsNavigation": true,
        "Definition": "public string PrimaryCurrency { get; set; }",
        "Initializer": "PrimaryCurrency = value.PrimaryCurrency == null ? null : value.PrimaryCurrency.Name",
        "Finalizer": "PrimaryCurrency = PrimaryCurrency == null ? null : context.CurrencyByKey[PrimaryCurrency]"
      }
    ]
  },
  {
    "HasKey": true,
    "Name": "Currency",
    "Properties": [
      {
        "IsNavigation": false,
        "Definition": "public string Name { get; set; }",
        "Initializer": "Name = value.Name",
        "Finalizer": "Name = Name"
      },
      {
        "IsNavigation": false,
        "Definition": "public string Symbol { get; set; }",
        "Initializer": "Symbol = value.Symbol",
        "Finalizer": "Symbol = Symbol"
      },
      {
        "IsNavigation": false,
        "Definition": "public SymbolPosition SymbolPosition { get; set; }",
        "Initializer": "SymbolPosition = value.SymbolPosition",
        "Finalizer": "SymbolPosition = SymbolPosition"
      },
      {
        "IsNavigation": false,
        "Definition": "public string Description { get; set; }",
        "Initializer": "Description = value.Description",
        "Finalizer": "Description = Description"
      }
    ]
  },
  {
    "HasKey": false,
    "Name": "Ledger",
    "Properties": [
      {
        "IsNavigation": false,
        "Definition": "public List<CurrencyData> Currencies { get; set; }",
        "Initializer": "Currencies = value.Currencies.Select(x => new CurrencyData(x)).ToList()",
        "Finalizer": "Currencies = new List<Currency>(Currencies.Select(x => x.Back(context)))"
      },
      {
        "IsNavigation": false,
        "Definition": "public List<AccountData> Accounts { get; set; }",
        "Initializer": "Accounts = value.Accounts.Select(x => new AccountData(x)).ToList()",
        "Finalizer": "Accounts = new List<Account>(Accounts.Select(x => x.Back(context)))"
      },
      {
        "IsNavigation": false,
        "Definition": "public List<BatchDescriptor> Batches { get; set; }",
        "Initializer": "Batches = value.Batches",
        "Finalizer": "Batches = Batches"
      }
    ]
  }
]